sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/SparkDateTimeUtils.scala

```diff
@@ -305,21 +305,35 @@ trait SparkDateTimeUtils {
       (segment == 0 && digits >= 4 && digits <= maxDigitsYear) ||
         (segment != 0 && digits > 0 && digits <= 2)
     }
-    if (s == null || s.trimAll().numBytes() == 0) {
+    if (s == null) {
       return None
     }
+
     val segments: Array[Int] = Array[Int](1, 1, 1)
     var sign = 1
     var i = 0
     var currentSegmentValue = 0
     var currentSegmentDigits = 0
-    val bytes = s.trimAll().getBytes
+    val bytes = s.getBytes
     var j = 0
+    var strEndTrimmed = bytes.length
```

Comment:
I like the previous name, it was more universal. e.g. `strLen` or `len`.

Reply:
How about
```
val strLen = bytes.length
var strEndTrimmed = strLen
```
We keep `strLen` immutable and reuse it later. `strEndTrimmed` used as mutable variable.

Reply:
`bytes.length` is very cheap and clear. It doesn't seem helpful to create a variable for it.

Reply:
Yeah, I also don't think that we need two variables. I can call it strLen or strLenTrimmed or anything along those lines, if that sounds better to you guys.

Reply:
Btw, we don't need strLen/bytes.length at all after we get strEndTrimmed var. We don't care about anything after strEndTrimmed in this function.

Reply:
I'm OK keep it unchanged here too.